profiles/models.py

```diff
@@ -51,12 +52,53 @@ def __ne__(self, other: Any) -> bool:
         return not self.__eq__(other)
 
 
+class Affiliation(db.Model):
+    _id = db.Column(db.Integer, name='id', primary_key=True)
+    department = db.Column(db.Text())
+    organisation = db.Column(db.Text(), nullable=False)
+    city = db.Column(db.Text(), nullable=False)
+    region = db.Column(db.Text())
+    _country = db.Column(db.String(2), name='country', nullable=False)
+    restricted = db.Column(db.Boolean(), nullable=False)
+    profile_id = db.Column(db.String(ID_LENGTH), db.ForeignKey('profile.id'))
+    profile = db.relationship('Profile', back_populates='affiliations')
+    position = db.Column(db.Integer())
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, country: Country, organisation: str, department: str = None,
+                 city: str = None, region: str = None, restricted: bool = False) -> None:
```

Comment:
More matches ORCID's structure than our API's. `country`, `region` and `city` could form an `Address`.